src/compilers/v1/compiler_v1.cpp

```diff
@@ -0,0 +1,507 @@
+
+    // Inherited via Compiler
+    virtual ~CompilerV1() = default;
+    virtual bool Initialize(ShaderCache &cache) final;
+    virtual std::unique_ptr<ProgramBinary> Compile(CompileArgs const& args, Logger const& logger) const final;
+    virtual std::unique_ptr<ProgramBinary> Link(LinkerArgs const& args, Logger const& logger) const final;
+    virtual std::unique_ptr<ProgramBinary> Load(const void *data, size_t size) const final;
+    virtual std::unique_ptr<CompiledDxil> GetKernel(const char *name, ProgramBinary const& obj, CompiledDxil::Configuration const *, Logger const *logger) const final;
+    virtual std::byte * CopyWorkProperties(std::byte *WorkPropertiesBuffer, WorkProperties const& props) const final;
+    virtual size_t GetWorkPropertiesChunkSize() const final;
+    virtual uint64_t GetVersionForCache() const final;
+};
+
+class ProgramBinaryV1 : public ProgramBinary
+{
+public:
+    using unique_ptr = std::unique_ptr<clc_object, void(*)(clc_object*)>;
+
+private:
+     unique_ptr m_Object;
+
+public:
+    ProgramBinaryV1(unique_ptr obj);
+    clc_object *GetRaw() const { return m_Object.get(); }
+
+    // Inherited via ProgramBinary
+    virtual ~ProgramBinaryV1() = default;
+    virtual bool Parse(Logger const *logger) final;
+    virtual size_t GetBinarySize() const final;
+    virtual const void *GetBinary() const final;
+};
+
+class CompiledDxilV1 : public CompiledDxil
+{
+public:
+    using unique_ptr = std::unique_ptr<clc_dxil_object, void(*)(clc_dxil_object*)>;
+
+private:
+    unique_ptr m_Object;
+
+public:
+    CompiledDxilV1(ProgramBinaryV1 const& parent, unique_ptr obj);
+    clc_dxil_object *GetRaw() const { return m_Object.get(); }
+
+    // Inherited via CompiledDxil
+    virtual ~CompiledDxilV1() = default;
+    virtual size_t GetBinarySize() const final;
+    virtual const void *GetBinary() const final;
+    virtual void *GetBinary() final;
+};
+
+static clc_logger ConvertLogger(Logger const& logger)
+{
+    auto log = [](void *ctx, const char *msg) { static_cast<Logger*>(ctx)->Log(msg); };
+    clc_logger ret;
+    ret.error = log;
+    ret.warning = log;
+    ret.priv = (void*)&logger;
+    return ret;
+}
+
+#if 0
+#endif
+
+CompilerV1::CompilerV1(XPlatHelpers::unique_module compiler)
+    : m_Compiler(std::move(compiler))
+{
+#define GET_FUNC(Member, api) Member = m_Compiler.proc_address<decltype(&api)>(#api)
+    GET_FUNC(CreateContext, clc_context_new);
+    GET_FUNC(SerializeContext, clc_context_serialize);
+    GET_FUNC(DeserializeContext, clc_context_deserialize);
+    GET_FUNC(FreeSerializedContext, clc_context_free_serialized);
+    GET_FUNC(FreeContext, clc_free_context);
+    GET_FUNC(CompileImpl, clc_compile);
+    GET_FUNC(LinkImpl, clc_link);
+    GET_FUNC(FreeSpirv, clc_free_object);
+    GET_FUNC(GetKernelImpl, clc_to_dxil);
+    GET_FUNC(FreeDxil, clc_free_dxil_object);
+    GET_FUNC(GetCompilerVersion, clc_compiler_get_version);
+#undef GET_FUNC
+
+    if (!CreateContext ||

```

Comment:
How come there are more GET_FUNC invocations above than entrypoints validated here?

Reply:
Older versions of the `clglon12compiler.dll` didn't have the context serialization/deserialization or version exports, so those are treated as optional for the v1 compiler. I'll add a comment.